lib/puppet/provider/dsc_base_provider/dsc_base_provider.rb

```diff
@@ -278,11 +278,15 @@ def should_to_resource(should, context, dsc_invoke_method)
     end
     resource[:dsc_invoke_method] = dsc_invoke_method
 
-    # Because Puppet adds all of the modules to the LOAD_PATH we can be sure that the appropriate module lives here
+    # Because Puppet adds all of the modules to the LOAD_PATH we can be sure that the appropriate module lives here during an apply;
     # PROBLEM: This currently uses the downcased name, we need to capture the module name in the metadata I think.
+    # During a Puppet agent run, the code lives in the cache so we can use the file expansion to discover the correct folder.
     root_module_path = $LOAD_PATH.select { |path| path.match?(%r{#{resource[:dscmeta_module_name].downcase}/lib}) }.first
-    resource[:vendored_modules_path] = File.expand_path(root_module_path + '/puppet_x/dsc_resources')
-    # resource[:vendored_modules_path] = File.expand_path(Pathname.new(__FILE__).dirname + '../../../' + 'puppet_x/dsc_resources')
+    resource[:vendored_modules_path] = if root_module_path.nil?
+                                         File.expand_path(Pathname.new(__FILE__).dirname + '../../../' + 'puppet_x/dsc_resources')
```

Comment:
Does this collide? In other words, if two modules vendor the same DSC resources for some reason, will they both pluginsync to the same path?

(Not a blocker, that's a problem with all current Puppet modules already.)

Reply:
I think so, yes. with the DSC Builder they'll also generate the same puppet resource names. This has already been identified as an issue for old vs new DSC modules, too. As you said, this is a wide-spread problem, and not something we can solve at this level.

Reply:
@binford2k, @DavidS: yeah, we confirmed yesterday that the new resources can't live side-by-side with the old puppetlabs-dsc module for this reason.

We could maybe change the folder path they're vendored to from `puppet_x/DscResources` to `puppet_x/dsc_resources` to resolve that particular issue, if we want them to be usable side-by-side. Though I don't think we'll have any way to prevent this from the initial issue you raised, @binford2k.